1481 2024 access control rework (#2366)
kanidm/kanidm: book/src/accounts/people.md

# People Accounts

A person represents a human's account in Kanidm. The majority of your users will be a person who

> will use this account in their daily activities. These entries may contain personaly identifying
> information that _is_ considered by kani to be sensitive. Because of this, there are default limits

Reviewer: ```suggestion
will use this account in their daily activities. These entries may contain personally identifying
information that _is_ considered by Kanidm to be sensitive. Because of this, there are default limits
```

to who may access these data.

## Creating Person Accounts

Members of the `idm_people_admins` group have the privileges to create new persons in the system. By
default `idm_admin` has this permission.

```bash
kanidm login --name idm_admin
kanidm person create demo_user "Demonstration User" --name idm_admin
kanidm person get demo_user --name idm_admin
```

Kanidm allows person accounts to include personally identifying attributes, such as their legal name
and email address.

Initially, a person does not have these attributes. If desired, a person may be modified to have
these attributes.

```bash
# Note, both the --legalname and --mail flags may be omitted
kanidm person update demo_user --legalname "initial name" --mail "initial@email.address"
```

You can also use anonymous to view accounts - note that you won't see certain fields due to the

> limits of the access control anonymous access profile.

Reviewer: ```suggestion
limits of the anonymous access control profile.
```

<!-- deno-fmt-ignore-start -->

{{#template ../templates/kani-warning.md
imagepath=images
title=Warning!
text=Persons may change their own displayname, name, and legal name at any time. You MUST NOT use these values as primary keys in external systems. You MUST use the `uuid` attribute present on all entries as an external primary key.
}}

<!-- deno-fmt-ignore-end -->

## Account Validity

Kanidm supports accounts that are only able to authenticate between a pair of dates and times; the
"valid from" and "expires" timestamps define these points in time. By default members of
`idm_people_admins` may change these values.

The account validity can be displayed with:

```bash
kanidm person validity show demo_user --name idm_admin
user: demo_user
valid after: any time
expire: never
```

```bash
kanidm person validity show demo_user --name idm_admin
valid after: 2020-09-25T21:22:04+10:00
expire: 2020-09-25T01:22:04+10:00
```

These datetimes are stored in the server as UTC, but presented according to your local system time
to aid correct understanding of when the events will occur.

You may set these time and date values in any timezone you wish (such as your local timezone), and

> the server will transform these to UTC. These time values are in iso8601 format, and you should

Reviewer: ```suggestion
the server will transform these to UTC. These time values are in ISO8601 format, and you should
```
